### PR TITLE
added native set support

### PR DESCRIPTION
**Pull request by googley42**
this resolves https://github.com/zio/zio-dynamodb/issues/59

### CODE REVIEW

File: dynamodb/src/main/scala/zio/dynamodb/Codec.scala [hunk 1]
```diff
@@ -572,11 +632,82 @@ private[dynamodb] object Codec {
       case av                        => Left(s"unable to decode $av as a list")
     }
 
+    private def setDecoder[A](s: Schema[A]): Decoder[Set[A]] = {
+      def nativeStringSetDecoder[A]: Decoder[Set[A]] = {
+        case AttributeValue.StringSet(stringSet) =>
+          Right(stringSet.asInstanceOf[Set[A]])
+        case av                                  =>
+          Left(s"Error: expected a set but found '$av'")
```
Comment (nhyne) [inline dynamodb/src/main/scala/zio/dynamodb/Codec.scala:640]: Do we want the error here to say `... expected a string set but ...`?

File: dynamodb/src/test/scala/zio/dynamodb/codec/SetCodecSpec.scala [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  }
+  final case class HasJavaBigDecimalSet(set: Set[java.math.BigDecimal])
+  object HasJavaBigDecimalSet    {
+    implicit val schema: Schema[HasJavaBigDecimalSet] = DeriveSchema.gen[HasJavaBigDecimalSet]
+  }
+  final case class HasBigIntSet(set: Set[BigInt])
+  object HasBigIntSet            {
+    implicit val schema: Schema[HasBigIntSet] = DeriveSchema.gen[HasBigIntSet]
+  }
+  final case class HasJavaBigIntegerSet(set: Set[BigInt])
```
Comment (nhyne) [inline dynamodb/src/test/scala/zio/dynamodb/codec/SetCodecSpec.scala:25]: It looks like this and `HasBigIntSet` are the same?

**Review by nhyne — APPROVED** [review]
Nice work!

File: dynamodb/src/main/scala/zio/dynamodb/Codec.scala (hunk 1 above)
Comment (googley42) [reply dynamodb/src/main/scala/zio/dynamodb/Codec.scala:640]: :+1:

File: dynamodb/src/test/scala/zio/dynamodb/codec/SetCodecSpec.scala (hunk 2 above)
Comment (googley42) [reply dynamodb/src/test/scala/zio/dynamodb/codec/SetCodecSpec.scala:25]: :+1:

**Pull request merged**